CMake/FindCUDNN.cmake

```diff
@@ -23,15 +23,27 @@ find_library( CUDNN_LIBRARIES
             ${CUDNN_TOOLKIT_ROOT_DIR}
       PATH_SUFFIXES lib lib64 lib/x64 lib/x86 targets/aarch64-linux
     )
+
+find_path(CUDNN_INCLUDE_DIR cudnn.h
+    PATHS ${CUDNN_ROOT_DIR} ${CUDA_TOOLKIT_ROOT_DIR}
```

Comment:
The CUDA / CUDNN environment variables do not seem to be well standardized. Other libraries that I've worked with will check a few different environment variables. 
For example: `cuda_environs = ['CUDA_HOME', 'CUDA_PATH', 'CUDA_SDK_ROOT_DIR', 'CUDAHOME']`

Reply:
Agreed, But I am trying to set the standard, at least in fletch to be CUDNN_TOOLKIT_ROOT_DIR
This is a CMAKE variable we expose to the user, so I would rather code to it rather than CUDNN_ROOT_DIR

Reply:
So, looking around other projects, caffe2 and caffe seems to the standard `CUDNN_ROOT_DIR`. 

From caffe2

```
cmake/Modules/FindCuDNN.cmake:#  CUDNN_ROOT_DIR:            Base directory where all cuDNN components are found
cmake/Modules/FindCuDNN.cmake:set(CUDNN_ROOT_DIR "" CACHE PATH "Folder contains NVIDIA cuDNN")
cmake/Modules/FindCuDNN.cmake:    PATHS ${CUDNN_ROOT_DIR} ${CUDA_TOOLKIT_ROOT_DIR}
cmake/Modules/FindCuDNN.cmake:    PATHS ${CUDNN_ROOT_DIR} ${CUDA_TOOLKIT_ROOT_DIR}
cmake/Modules/FindCuDNN.cmake:  mark_as_advanced(CUDNN_ROOT_DIR CUDNN_LIBRARY CUDNN_INCLUDE_DIR)
```

From caffe

```cmake
cmake/Cuda.cmake:  set(CUDNN_ROOT "" CACHE PATH "CUDNN root folder")
cmake/Cuda.cmake:            PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT} ${CUDA_TOOLKIT_INCLUDE}
cmake/Cuda.cmake:                             PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT} ${CUDNN_INCLUDE} ${__libpath_hist}
cmake/Cuda.cmake:    mark_as_advanced(CUDNN_INCLUDE CUDNN_LIBRARY CUDNN_ROOT)
```

pytorch also seems to use `CUDNN_ROOT_DIR`


I don't see reference to any specific CUDNN variable in OpenCV or Tensorflow.  I've never seen `CUDNN_TOOLKIT_ROOT_DIR` used anywhere other than here, so perhaps we should follow the existing standard and use `CUDNN_ROOT_DIR` instead?